packages/backend/src/App.ts

```diff
@@ -789,25 +772,11 @@ export default class App {
     }
 
     private async initSlack(expressApp: Express) {
-        let aiAgentService: AiAgentService | undefined;
-        try {
-            aiAgentService = this.serviceRepository.getAiAgentService();
-        } catch (e) {
-            // ai agent service is not available for OSS
-        }
+        const slackClient = this.clients.getSlackClient();
+        await slackClient.start(expressApp);
 
-        const slackClient = this.slackClientFactory({
-            lightdashConfig: this.lightdashConfig,
-            analytics: this.analytics,
-            serviceRepository: this.serviceRepository,
-            models: this.models,
-            clients: this.clients,
-        });
-        await slackClient.start(
-            expressApp,
-            this.serviceRepository.getUnfurlService(),
-            aiAgentService,
-        );
+        const slackService = this.serviceRepository.getSlackService();
+        slackService.setupEventListeners();
```

Comment:
sets up listeners - after initializing the client

Reply:
so a revert of the revert with a fix of the order of initialisation?